### PR TITLE
Chore: use nyc instead of istanbul

### PR DESCRIPTION
**Pull request by mysticatea**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Other, please explain

**What changes did you make? (Give an overview)**

This PR switch from `istanbul` to [nyc] the official successor of `istanbul` CLI.

This is needed in order to use newer ES syntaxes.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular.

[nyc]: https://github.com/istanbuljs/nyc

### CODE REVIEW

File: Makefile.js [hunk 1]
```diff
@@ -84,12 +84,7 @@ const NODE = "node ", // intentional extra space
  * @private
  */
 function getTestFilePatterns() {
-    return ls("tests/lib/").filter(pathToCheck => test("-d", `tests/lib/${pathToCheck}`)).reduce((initialValue, currentValues) => {
-        if (currentValues !== "rules") {
-            initialValue.push(`"tests/lib/${currentValues}/**/*.js"`);
-        }
-        return initialValue;
-    }, ["\"tests/lib/rules/**/*.js\"", "\"tests/lib/*.js\"", "\"tests/bin/**/*.js\"", "\"tests/tools/**/*.js\""]).join(" ");
+    return ["\"tests/lib/**/*.js\"", "\"tests/bin/**/*.js\"", "\"tests/tools/**/*.js\""].join(" ");
```
Comment (mysticatea) [inline Makefile.js]: I think this is same.

- `tests/lib/*.js` + `tests/lib/rules/**/*.js` + `tests/lib/${x}/**/*.js` (x is the directories on `tests/lib` except `tests/lib/rules`) = `tests/lib/**/*.js`.

Comment (platinumazure) [reply Makefile.js]: I assume `"tests/lib/*.js"` is covered by the new pattern as well?

Comment (mysticatea) [reply Makefile.js]: Yes!

Comment (aladdin-add) [inline Makefile.js]: maybe it could be simplified to `return "tests/{lib,bin,tools}/**/*.js";`.

Comment (mysticatea) [reply Makefile.js]: Good catch!

**Pull request merged**